nfs: add support for running SSSD as a sidecar (#10676)
rook/rook: Documentation/Storage-Configuration/NFS/nfs-security.md

---
title: Security
---

## User ID management

!!! attention

>     This feature is experimental, and we do not guarantee it is bug-free, nor will

Reviewer: bug free? :)

    we support upgrades to future versions.

!!! attention
    Some configurations of this feature may break the ability to mount NFS storage to pods via PVCs.

### ID management via SSSD

[SSSD](https://sssd.io) is the System Security Services Daemon. It can be used to provide user ID
management from a  number of sources including LDAP, Active Directory, and FreeIPA. Currently, only
LDAP has been tested.

!!! attention
    The Ceph container image must have the `sssd-client` package installed to support SSSD. This

>     package is added to `quay.io/ceph/ceph` in v17.2.4. For lower Ceph versions you may

Reviewer: how about mentioning versions as if there are already newer versions available?
```suggestion
    package is included in `quay.io/ceph/ceph` in v17.2.4 or newer. For lower Ceph versions you may
```

---
title: Security
---

## User ID management

> !!! attention

Reviewer: We only have attention notes in this section. Is there some summary of ID management to include as well?

Author: Good point. I'll add a simple overview.

>     This feature is experimental and will not support upgrades to future versions.

Reviewer: Do we know of something that would make upgrades in the future difficult? If it might work, perhaps we soften the wording.
```suggestion
    This feature is experimental and may not support upgrades to future versions.
```

Author: Softening the wording seems good. I think it's likely we will be able to upgrade just fine, but there could be some breaking API changes if we discover something critical.

LDAP has been tested.

!!! attention
    The Ceph container image must have the `sssd-client` package installed to support SSSD. This
    package is included in `quay.io/ceph/ceph` in v17.2.4 and newer. For older Ceph versions you may
    build your own Ceph image which adds `RUN yum install sssd-client && yum clean all`.

#### **SSSD configuration**

SSSD requires a configuration file in order to configure its connection to the user ID management
system (e.g., LDAP). The file follows the `sssd.conf` format documented in its
[man pages](https://linux.die.net/man/5/sssd.conf).

Methods of providing the configuration file are documented in the
[NFS CRD security section](../../CRDs/ceph-nfs-crd.md#security).

The SSSD configuration file may be omitted from the CephNFS spec if desired. In this case, Rook will
not set the `/etc/sssd/sssd.conf` in any way. This allows you to manage the `sssd.conf` file
yourself however you wish. For example, you may build it into your custom Ceph container image, or
use the [Vault agent injector](https://www.vaultproject.io/docs/platform/k8s/injector) to securely

> add the file via annotations on the CephNFS spec (passed to the NFS server pods).

Author: Does this section seem adequate to explain how to configure things?
1. refer to CRD docs
2. some more info if you want to specify your own

Reviewer: Yes this is helpful to have the pointers for how they might choose to do it.

---
title: Security
---

## User ID mapping
User ID mapping allows the NFS server to map connected NFS client IDs to a different user domain,
allowing NFS clients to be associated with a particular user in your organization. For example,
users stored in LDAP can be associated with NFS users and vice versa.

> 

Author: In addition to this overview, I realized I'd ben using "ID **management**" when I should have been using "ID **mapping**".

!!! attention
    This feature is experimental and may not support upgrades to future versions.

!!! attention
    Some configurations of this feature may break the ability to
    [mount NFS storage to pods via PVCs](./nfs-csi-driver.md#attaching-an-export-to-a-pod).

>     The NFS CSI driver may not be able to mount exports for pods when ID mapping is configured.

Author: Good addition?
